crates/cow-amm/src/signing.rs

```diff
@@ -0,0 +1,92 @@
+//! Signing utilities for CoW AMM interactions.
+//!
+//! This module provides helpers for creating signatures and interactions
+//! required by CoW AMM contracts when participating in CoW Protocol
+//! settlements.
+
+/// GPv2Order-specific signing utilities.
+///
+/// CoW Protocol uses GPv2Order structs for order representation. The EIP-712
+/// signing has a mismatch between on-chain storage (using `bytes32` for certain
+/// fields) and the type string used for signing (using `string` for
+/// human-readable wallet displays).
+///
+/// See: <https://github.com/cowprotocol/contracts/blob/v1.1.2/src/contracts/libraries/GPv2Order.sol#L26-L48>
+pub mod gpv2_order {
```

Comment:
What's the reason for this additional module? I don't think it's needed but if it was I'd suggest to move the module definition into `lib.rs` to avoid "wasting" 1 level of indentation in the entire file here.